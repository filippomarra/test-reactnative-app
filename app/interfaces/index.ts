import Story from '../models';

export interface ApiProps {
    setLoading: (loading: boolean) => void;
    setStories: (stories: Story[]) => void;
}

export interface AppHeaderProps {
    stories: Story[];
    setStories: (stories: Story[]) => void;
}

export interface ModalProps {
    visible: boolean;
    hideDialog: () => void;
    story: Story;
}

export interface StoryItemProps {
    index: number;
    showDialog: (story: Story) => void;
    story: Story;
}

export interface StoryScrollViewProps {
    isLoading: boolean;
    showDialog: (story: Story) => void;
    stories: Story[];
}
