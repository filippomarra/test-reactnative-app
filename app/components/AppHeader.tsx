import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { Header, Icon } from '@rneui/themed';

import { APP_TITLE } from '../constants';
import { AppHeaderProps } from '../interfaces';
import { styles } from '../styles';

import { sortByKey } from './utils';

export const AppHeader = (props: AppHeaderProps) => {
    const { stories, setStories } = props;

    return (
        <Header
            centerComponent={{
                text: APP_TITLE,
                style: styles.header,
            }}
            rightComponent={
                <View style={styles.headerRight}>
                    <TouchableOpacity
                        onPress={() => setStories(sortByKey(stories, 'name'))}
                        testID={'sortButton'}>
                        <Icon name="sort" color="white" size={32} />
                        <Text style={styles.sortText}> SORT</Text>
                    </TouchableOpacity>
                </View>
            }
        />
    );
};
