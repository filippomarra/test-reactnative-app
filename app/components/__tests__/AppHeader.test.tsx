import 'react-native';
import React from 'react';

import { fireEvent, render } from '@testing-library/react-native';

import { AppHeaderProps } from '../../interfaces';

import { AppHeader } from '../AppHeader';

const defaultProps: AppHeaderProps = {
    setStories: jest.fn(),
    stories: [],
};

describe('AppHeader component tests', () => {
    it('renders correctly with default props', () => {
        expect(
            render(<AppHeader {...defaultProps} />).toJSON(),
        ).toMatchSnapshot();
    });

    describe('When the sort button is not pressed', () => {
        it('should not invoke the setStories function', () => {
            const setStoriesSpy = jest.fn();
            render(<AppHeader {...defaultProps} setStories={setStoriesSpy} />);
            expect(setStoriesSpy).not.toHaveBeenCalled();
        });
    });

    describe('When the sort button is pressed', () => {
        it('should invoke the setStories function', () => {
            const setStoriesSpy = jest.fn();
            const { getByTestId } = render(
                <AppHeader {...defaultProps} setStories={setStoriesSpy} />,
            );
            const sortButton = getByTestId('sortButton');
            fireEvent(sortButton, 'onPress');
            expect(setStoriesSpy).toHaveBeenCalled();
        });
    });
});
