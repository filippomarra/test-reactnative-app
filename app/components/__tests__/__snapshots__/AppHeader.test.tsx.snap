// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader component tests renders correctly with default props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2089dc",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
  testID="headerContainer"
>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "top",
        "right",
      ]
    }
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#fff",
            "fontFamily": "TitilliumWeb-Bold",
            "fontSize": 37,
            "paddingTop": 13,
          }
        }
        text="PAGOPA TEST"
      >
        PAGOPA TEST
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "fontFamily": "TitilliumWeb-Bold",
            "paddingTop": 13,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="sortButton"
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                Object {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              testID="RNE__ICON__CONTAINER_ACTION"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 32,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="RNE__ICON__Component"
                >
                  î…¤
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "TitilliumWeb-Bold",
              }
            }
          >
             SORT
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
